mission_control/navigator_missions/nav_missions/pinger.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+from __future__ import division
+import txros
+import std_srvs.srv
+import numpy as np
+import tf
+import tf.transformations as trns
+from navigator_msgs.msg import ShooterDoAction, ShooterDoActionGoal
+from navigator_msgs.srv import CameraToLidarTransform, CameraToLidarTransformRequest, FindPinger, FindPingerRequest, SetFrequency, SetFrequencyRequest
+from geometry_msgs.msg import Point
+from std_srvs.srv import SetBool, SetBoolRequest
+from twisted.internet import defer
+from image_geometry import PinholeCameraModel
+from visualization_msgs.msg import Marker,MarkerArray
+import navigator_tools
+
+class PingerMission:
+    OBSERVE_DISTANCE_METERS = 12
+    GATE_CROSS_METERS = 10
+    FREQ = 35000
+    LISTEN_TIME = 15
+
+    def __init__(self, navigator):
+        self.navigator = navigator
+        self.listen_client = self.navigator.nh.get_service_client('/hydrophones/set_listen', SetBool)
+        self.pinger_client = self.navigator.nh.get_service_client('/hydrophones/find_pinger', FindPinger)
+        self.reset_client = self.navigator.nh.get_service_client('/hydrophones/set_freq', SetFrequency)
+        self.markers_pub = self.navigator.nh.advertise("/pinger/debug_marker",MarkerArray)
+        self.negate = False
+
+    @txros.util.cancellableInlineCallbacks
+    def reset_freq(self):
+        yield self.reset_client(SetFrequencyRequest(frequency=self.FREQ))
+
+    #@txros.util.cancellableInlineCallbacks
+    def start_listen(self):
+        return self.listen_client(SetBoolRequest(data=True))
+
+    #@txros.util.cancellableInlineCallbacks
+    def stop_listen(self):
+        return self.listen_client(SetBoolRequest(data=False))
+
+    @txros.util.cancellableInlineCallbacks
+    def get_objects(self):
+        self.gate_1 = yield self.navigator.database_query("Gate_1")
+        assert self.gate_1.found, "Not found"
+        self.gate_1_pos = navigator_tools.rosmsg_to_numpy(self.gate_1.objects[0].position)[:2]
+        self.gate_2 = yield self.navigator.database_query("Gate_2")
+        assert self.gate_2.found, "Not found"
+        self.gate_2_pos = navigator_tools.rosmsg_to_numpy(self.gate_2.objects[0].position)[:2]
+        self.gate_3 = yield self.navigator.database_query("Gate_3")
+        assert self.gate_3.found, "Not found"
+        self.gate_3_pos = navigator_tools.rosmsg_to_numpy(self.gate_3.objects[0].position)[:2]
+
+    def get_observation_poses(self):
+        self._get_perp(self.gate_1_pos, self.gate_3_pos)
+
+        pose = yield self.navigator.tx_pose()
```

Comment:
With `@property` decorators (like `navigator.tx_pose`'s) you don't need the `( )` at the end (it's treated like a class function), and it should throw an error if you do - but I think since that function is double decorated, nothing bad happens.